File: dev-docs/snippets/events/blobs-upload-failed.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when a queued blob upload fails (may still retry).
export function blobsUploadFailed(client: JsBaoClient) {
  // #region example
  client.on("blobs:upload-failed", (payload) => {
    console.log("failed", payload.blobId, payload.lastError, "retry:", payload.willRetry);
  });
  // #endregion example
}

File: dev-docs/snippets/events/blobs-upload-failed.swift
import JsBaoClient

// Fires when a queued blob upload fails (may still retry).
func blobsUploadFailed(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.blobsUploadFailed) { (event: BlobUploadFailedEvent) in
    // Full record matching JS: lastError? / attempts / nextAttemptAt / willRetry / …
    print("failed", event.blobId, event.lastError ?? "", "retry:", event.willRetry)
  }
  // #endregion example
  _ = sub
}
